src/datasources.js

```diff
@@ -300,7 +307,14 @@ define([
         }
         function getNodes(source, path, info) {
             var nodes = _.chain(source && source.structure)
-                .map(node(source, path, info))
+                .map(function (item, id) {
+                    if (item.merge) {
```

Comment:
Correct, merge is for the UI. I'll think about how to improve the documentation. Do you want to see that in this PR? The schema transformation is complicated, and the docstring at the top of the file is supposed to explain how it's interpreted. I wonder if the schema itself could be simplified so the code was easier to understand?